### ISSUE TITLE
Symfony\Component\Stopwatch\Tests\StopwatchEventTest::testEnsureStopped is volatile

### ISSUE THREAD

**Issue by stof** [symfony/symfony#14444]
See https://travis-ci.org/symfony/symfony/jobs/59546578

**Comment by dosten** [comment]
I cannot see any error in that job, here yes https://travis-ci.org/symfony/symfony/jobs/59554714.
```
PHPUnit 4.5.0 by Sebastian Bergmann and contributors.

Configuration read from /home/travis/build/symfony/symfony/phpunit.xml.dist

................................................................. 65 / 66 ( 98%)
.

Time: 9.93 seconds, Memory: 17.40Mb

OK (66 tests, 71 assertions)

Running src/Symfony/Component/Stopwatch tests
PHPUnit 4.5.0 by Sebastian Bergmann and contributors.

Configuration read from /home/travis/build/symfony/symfony/phpunit.xml.dist

.........FF......F.....

Time: 11.86 seconds, Memory: 17.34Mb

There were 3 failures:

1) Symfony\Component\Stopwatch\Tests\StopwatchEventTest::testEnsureStopped
Failed asserting that 422 matches expected 300.

/home/travis/build/symfony/symfony/src/Symfony/Component/Stopwatch/Tests/StopwatchEventTest.php:134

2) Symfony\Component\Stopwatch\Tests\StopwatchEventTest::testStartTime
Failed asserting that false is true.

/home/travis/build/symfony/symfony/src/Symfony/Component/Stopwatch/Tests/StopwatchEventTest.php:145

3) Symfony\Component\Stopwatch\Tests\StopwatchTest::testStop
Failed asserting that 228 matches expected 200.

/home/travis/build/symfony/symfony/src/Symfony/Component/Stopwatch/Tests/StopwatchTest.php:79

FAILURES!
Tests: 23, Assertions: 37, Failures: 3.
```
we can increase the delta in `StopwatchTest` to 35, but increasing the delta in `StopwatchEventTest` to 150 is too much, right?

**Comment by stof** [comment]
@dosten a volatile test means it fails randomly (depending on the load of the Travis servers for instance)

**Comment by dosten** [comment]
@stof yes, i know. I'm saying that the failing assertions have a 4th parameter named $delta. Ex:
```php
public function testEnsureStopped()
{
    // this also test overlap between two periods
    $event = new StopwatchEvent(microtime(true) * 1000);
    $event->start();
    usleep(100000);
    $event->start();
    usleep(100000);
    $event->ensureStopped();
    $this->assertEquals(300, $event->getDuration(), null, self::DELTA); // delta = 37
}
```
In the example, phpunit assert as equals values between [263, 337].
The error is: "Failed asserting that 422 matches expected 300.", so the Travis server has 122 miliseconds of delay (in this case).
What i say in my previous comment is that we can increase `self:DELTA` to 150, so phpunit will assert as equals values between [150, 450].
It's a possible solution?
Sorry for my english, i'm noob

**Comment by stof** [comment]
@dosten if we increase the delta too much, the test will be totally useless. So updating blindly the delta until we never see failures anymore is not the solution either. In the end, we would not detect real bugs in the logic anymore if we allow too much

**Comment by dosten** [comment]
Same problem here `Symfony\Component\Process\Tests\SigchildEnabledProcessTest::testIdleTimeoutNotExceededWhenOutputIsSent` i think

**Comment by marcojanssen** [comment]
@stof considering both fixes have been merged i assume this ticket can be closed?

**Comment by WouterJ** [comment]
@marcojanssen the first ticket is not relevant to this one (it just mentions that the build in that PR is failing because of the volatile test in this issue). The second ticket was rejected, as it was not testing the method.

Tldr: Nothing has been fixed around this afaics.

**Comment by fabpot** [comment]
Closing as there is no point in keeping those tickets around. We either fix the problem or we don't care about them, we don't need reminders.
